datafusion/physical-expr/src/utils/guarantee.rs

```diff
@@ -334,21 +357,33 @@ impl<'a> ColOpLit<'a> {
             binary_expr.op(),
             binary_expr.right().as_any(),
         );
-
+        let guarantee = match op {
```

Comment:
Since operator except `=` and `!=` would be evaluate in predicate_expr, I put the conversion process here.

Reply:
👍